docs/running_lbann.rst

```diff
@@ -1,96 +1,434 @@
-             done by modifying the prototext directly or overriding
-             the options on the command line with, e.g., the
-             :code:`--data_filedir_train` and
-             :code:`--data_filedir_test` options.
-
-The following is an example invocation of LBANN on a machine using
-Slurm's :bash:`srun` as an MPI launcher. In the example command,
-a machine with 2 GPGPUs per node are available, 4 nodes will be used,
-:bash:`${LBANN_EXE}` is the path to the :code:`lbann` executable, and
-:bash:`${LBANN_MODEL_ZOO_DIR}` is the path to the :code:`model_zoo/` directory in
-either the source tree or the install tree. Note that the options
-passed to :bash:`srun` are not likely to be portable to other MPI
-launchers. The example will train Alexnet with SGD optimization on the
-Imagenet dataset for 5 epochs.
+    module load lbann
+
+.. note:: This will *not* work if LBANN has been built with
+  :bash:`scripts/build_lbann_lc.sh` or with the Spack developer build
+  process.
+
++ LBANN includes a modulefile that updates :bash:`PYTHONPATH`:
 
 .. code-block:: bash
 
-    srun -N4 --ntasks-per-node=2 \
-        ${LBANN_EXE} \
-        --model=${LBANN_MODEL_ZOO_DIR}/models/alexnet/alexnet.prototext \
-        --optimizer=${LBANN_MODEL_ZOO_DIR}/optimizers/opt_sgd.prototext \
-        --reader=${LBANN_MODEL_ZOO_DIR}/data_readers/data_reader_imagenet.prototext \
-        --num_epochs=5
-    
----------------------------------------------
-Using the Python interface for prototext
----------------------------------------------
-
-There is a python interface for generating model prototext
-files. Example Python scripts may be found in the
-:code:`scripts/proto/lbann/models` directory of the source
-repository. Running the Python script will generate a prototext that
-can be passed to the :code:`--model` option for LBANN.
+    module use <install directory>/etc/modulefiles
+    module load lbann-<version>
+
++ Directly manipulate :bash:`PYTHONPATH`:
 
 .. code-block:: bash
-                
-    python3 alexnet.py alexnet.prototext
-    <mpi-launcher> <mpi-options> \
-        lbann --model=alexnet.prototext <other-lbann-options>
 
-where :code:`<other-lbann-options>` are as documented
-:ref:`above <using-the-model-zoo>`, with optimizer and data reader
-prototexts coming from the appropriate :code:`model_zoo/` directories.
+    export PYTHONPATH=<install directory>/lib/python<version>/site-packages:${PYTHONPATH}
+
+Note that LBANN depends on the Protobuf Python package, which can be
+installed with:
+
+.. code-block:: bash
+
+    pip install protobuf
+
+If the user does not own the site-packages directory, then it may be
+necessary to pass the :bash:`--user` flag to pip.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Basic usage
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A typical workflow involves the following steps:
+
+1. Configuring LBANN model components (like the graph of
+:python:`Layer` s) and creating a :python:`Model`.
```

Comment:
Should this be "`Layer`s"? I don't think you want the space.

Reply:
Sphinx messes up without the space.

Reply:
Ah ok. Good to know.